### DISCUSSION TITLE
refreshing cache when using simplePagination

### DISCUSSION THREAD

**Discussion by tlthiem** [urql-graphql/urql#1258] [Q&A] [answered]
Hello,

I am pretty new to urql and wondering how i can support **refreshing** my feed while using simplePagination. My usecase is sort of an endless scrollable newsfeed. Fetching more data works great so far. But when I reset the pointer (offset to 0) to get the newest posts (aka refresh), the cache responds without asking the api for updates (which makes sense because it already saw this query with this particular offset (0) ).

Is there a way I can modify the resolver to refetch using network in this case? E.g. setting results as stale

I am using the cache to support instant updates on likes, etc. aswell so it would be great if the cache wouldnt get fully wiped.

I am currently using a wrapper around useQuery to fetchMore and trigger refreshes:

````ts
export default function usePagination({
  query,
  limit = 20,
  queryName,
  variables,
}: usePaginationArgs) {
  const [offset, setOffset] = React.useState(0);

  const [{ fetching, data, error }, reexecuteQuery] = useQuery({
    query,
    variables: { ...variables, limit, offset },
    requestPolicy: 'cache-and-network',
  });

  const fetchMore = React.useCallback(() => {
    setOffset((data && !error ? data[queryName].length : 0) );
  }, [data, error]);

  const refresh = () => {
    setOffset(0);
  };

  return [{ data, error, fetching }, fetchMore, refresh] as const;
}
``````

I cant be the only one facing this problem right?

I couldnt find a solution with refresh support yet.

I'd appreciate any kind of feedback.

Thanks

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

Well because you are using `cache-and-network` it _should_ ask for updates in the background, this means that you'll see an instant response from the cache when it's available in cache but the combination of your `offset` and `limit` will be refetched, this can be seen by you having data but the result still indicating a `fetching: true` state.

**Comment by tlthiem** [comment]
Hey, thanks for your reply,

Yes you are right, it looks like I am having a react problem here instead. Because if offset is already 0 no change is registered in state on next refresh trigger and therefore no update is triggered. Interestingly enough I cant trick react into thinking the value changed. Are state changes tracked in useQuery/urql ?. I am at a loss on how to implement/allow repetitive refresh triggering using only one (offset) state variable. If I'd add a flag It would mess with react update flows instead. Any tip on how to implement this. I feel like the solution is easy. Its just not getting to me :/

**Reply by JoviDeCroock** [reply]
You can use something like when the variables would remain the same.

```
export default function usePagination({
  query,
  limit = 20,
  queryName,
  variables,
}: usePaginationArgs) {
  const [offset, setOffset] = React.useState(0);

  const [{ fetching, data, error }, reexecuteQuery] = useQuery({
    query,
    variables: { ...variables, limit, offset },
    requestPolicy: 'cache-and-network',
  });

  const fetchMore = React.useCallback(() => {
    setOffset((data && !error ? data[queryName].length : 0) );
  }, [data, error]);

  const refresh = () => {
    if (offset === 0)
      reexecuteQuery({ requestPolicy: 'cache-and-network' });
    else
      setOffset(0)
  };

  return [{ data, error, fetching }, fetchMore, refresh] as const;
}
```

**Reply by tlthiem** [reply]
Thank you so much. This is it. You made my day :)

**Reply by JoviDeCroock** [reply]
No problem, glad I could help out :)
